include/http/request_handler.js

```diff
@@ -534,15 +535,26 @@ module.exports = function RequestHandlerModule(pb) {
      */
     RequestHandler.prototype.handleRequest = function(){
 
-        //fist things first check for public resource
+        //first things first check for public resource
         if (RequestHandler.isPublicRoute(this.url.pathname)) {
             return this.servePublicContent();
         }
 
         //check for session cookie
         var cookies = RequestHandler.parseCookies(this.req);
+        var siteObj = RequestHandler.sites[this.hostname];
+        var url = this.url.pathname;
         this.req.headers[pb.SessionHandler.COOKIE_HEADER] = cookies;
 
+        if(typeof(siteObj) !== 'undefined' && siteObj.hasOwnProperty('forceLocale') && siteObj.forceLocale) {
+            cookies.locale = cookies.locale || cookies[" locale"];
+            cookies.locale = cookies.locale || siteObj.defaultLocale;
+            if (checkRouteContainsLocale(url, siteObj)) {
+                var redirectLocation = pb.SiteService.getHostWithProtocol(this.hostname);
+                redirectLocation += '/' + cookies.locale + this.url.path;
+                return this.doRedirect(redirectLocation, pb.HttpStatus.MOVED_TEMPORARILY);
+            }
+        }
```

Comment:
On where the cookie gets set.  This is going to be the hard part for me.  We have a base plugin that contains an angular controller that is included on every page (in the header) and it will set the cookie by getting the locale that we have derived for the user.  Since this was a front end piece to be loaded by the html and did not live at the platform level, I was wondering how best it would be to submit that back to you all.